lib/web_ui/dev/test_runner.dart

```diff
@@ -448,69 +513,69 @@ class TestCommand extends Command<bool> with ArgUtils {
     timestampFile.writeAsStringSync(timestamp);
   }
 
+  Future<void> _buildTestsInParallel(
+      {List<FilePath> targets, bool forCanvasKit = false}) async {
+    final List<TestBuildInput> buildInputs = targets
+        .map((FilePath f) => TestBuildInput(f, forCanvasKit: forCanvasKit))
+        .toList();
+
+    final results = _pool.forEach(
+      buildInputs,
+      _buildTest,
+    );
+    await for (final bool isSuccess in results) {
+      if (!isSuccess) {
+        throw ToolException('Failed to compile tests.');
+      }
+    }
+  }
+
   /// Builds the specific test [targets].
   ///
   /// [targets] must not be null.
   ///
-  /// When building for CanvasKit we have to use a separate `build.canvaskit.yaml`
-  /// config file. Otherwise, `build.html.yaml` is used. Because `build_runner`
-  /// overwrites the output directories, we redirect the CanvasKit output to a
-  /// separate directory, then copy the files back to `build/test`.
-  Future<void> _buildTests({List<FilePath> targets, bool forCanvasKit}) async {
-    print(
-        'Building ${targets.length} targets for ${forCanvasKit ? 'CanvasKit' : 'HTML'}');
-    final String canvasKitOutputRelativePath =
-        path.join('.dart_tool', 'canvaskit_tests');
+  /// Uses `dart2js` for building the tests.
+  ///
+  /// When building for CanvasKit we have to use extra argument
+  /// `DFLUTTER_WEB_USE_SKIA=true`.
+  Future<bool> _buildTest(TestBuildInput input) async {
+    final targetFileName = input.target.relativeToWebUi
+        .replaceFirst('.dart', '.dart.browser_test.dart.js');
+    final String targetPath = path.join('build', targetFileName);
+
+    final io.Directory directoryToTarget = io.Directory(path.join(
+        environment.webUiBuildDir.path,
+        path.dirname(input.target.relativeToWebUi)));
+
+    if (!directoryToTarget.existsSync()) {
+      directoryToTarget.createSync(recursive: true);
+    }
+
     List<String> arguments = <String>[
-      'run',
-      'build_runner',
-      'build',
+      '--no-minify',
+      '--disable-inlining',
+      '--enable-asserts',
       '--enable-experiment=non-nullable',
-      'test',
+      '--no-sound-null-safety',
+      if (input.forCanvasKit) '-DFLUTTER_WEB_USE_SKIA=true',
+      '-O2',
       '-o',
-      forCanvasKit ? canvasKitOutputRelativePath : 'build',
-      '--config',
-      // CanvasKit uses `build.canvaskit.yaml`, which HTML Uses `build.html.yaml`.
-      forCanvasKit ? 'canvaskit' : 'html',
-      for (FilePath path in targets) ...[
-        '--build-filter=${path.relativeToWebUi}.js',
-        '--build-filter=${path.relativeToWebUi}.browser_test.dart.js',
-      ],
+      '${targetPath}', // target path.
```

Comment:
No need to interpolate `targetPath` it's already the exact string we want to pass to dart2js

Reply:
done